### PR TITLE
keep month, year, newmoonnumber columns correct in forecasts

### PR DESCRIPTION
**Pull request by emchristensen**
this should fix issue #110 
wrote a function to pull next 12 newmoondates from navy.mil, so when forecasting using PortalForecasts the month, year, and newmoonnumber columns remain in sync
PortalHindcasts had the same problem, fixed that too

### CODE REVIEW

**Comment by sdtaylor** [comment]
Couldn't we populate the `moon_dates.csv` file with every newmoon for a decade or so into the future? That seems easier than querying for it every time this runs.

**Comment by gmyenni** [comment]
We've decided to update tables in PortalData like `moon_dates.csv` as new data comes in, rather than project them out past existing data.

If you want a `moon_dates` that goes a decade into the future, we could have one here in portalPredictions. . .

**Comment by emchristensen** [comment]
Does looking it up every time slow down the code?  Populating a table for an arbitrary time period like a decade without some mechanism for updating it seems like creating problems for future people.  Maybe we can have Travis check it periodically and update it if necessary?

**Comment by sdtaylor** [comment]
I only suggest that because it's something that should never change once you get it once. If they can predict solar eclipses decades in advance then their new moon dates are probably solid.

File: models/model_functions.R [hunk 1]
```diff
@@ -14,6 +14,23 @@ get_moon_data <- function(){
   return(moons)
 }
```
Comment (sdtaylor) [inline models/model_functions.R:16]: Need a comment here such as
```
# Returns a data.frame of newmoons in the future in the same format of moons
# moons is the output of get_moon_data()
```

File: PortalForecasts.R [hunk 2]
```diff
@@ -12,14 +12,15 @@ forecast_date = Sys.Date()
 
 portalr::download_observations()
 moons = get_moon_data()
```
Comment (sdtaylor) [inline PortalForecasts.R:14]: add comment here like in portalhindcasts

```
# #get dates of 12 new moons following newmoon of interest
```

File: models/model_functions.R [hunk 3]
```diff
@@ -14,6 +14,23 @@ get_moon_data <- function(){
...
 
+get_future_moons <- function(moons){
+  #Get dates of future new moons from navy website
+  year = tail(moons$year,1)
```
Comment (sdtaylor) [inline models/model_functions.R]: change these to `most_recent_year` and `most_recent_month` to be more descriptive

**Review by sdtaylor — CHANGES_REQUESTED** [review]
This looks good. I just put some suggestions to make it easier to follow.

**Comment by emchristensen** [comment]
Ok @sdtaylor I tried to incorporate all your edits -- give it another look and see what you think

**Review by sdtaylor — APPROVED** [review]

**Pull request merged**